packages/block-library/src/list-item/style.native.scss

```diff
@@ -28,6 +28,11 @@
 	font-size: $editor-font-size;
 }
 
+.wp-block-list-item__list-item--default--dark {
+	color: $white;
+	font-size: $editor-font-size;
```

Comment:
Nitpick we can remove the `font-size` here as it would use the value from `wp-block-list-item__list-item--default`